components/view_image/view_image.tsx

```diff
@@ -20,170 +23,203 @@ import PopoverBar from './popover_bar';
 
 const KeyCodes = Constants.KeyCodes;
 
-export default class ViewImageModal extends React.PureComponent {
-    static propTypes = {
-
-        /**
-         * The post the files are attached to
-         */
-        postId: PropTypes.string,
-
-        /**
-         * The post the files are attached to
-         */
-        post: PropTypes.object,
-
-        /**
-         * Set whether to show this modal or not
-         */
-        show: PropTypes.bool.isRequired,
-
-        /**
-         * Function to call when this modal is dismissed
-         **/
-        onModalDismissed: PropTypes.func.isRequired,
-
-        /**
-         * List of FileInfo to view
-         **/
-        fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-
-        /**
-         * The index number of starting image
-         **/
-        startIndex: PropTypes.number,
-
-        canDownloadFiles: PropTypes.bool,
-        enablePublicLink: PropTypes.bool,
-        pluginFilePreviewComponents: PropTypes.arrayOf(PropTypes.object),
-    };
+export type Props = {
+
+    /**
+ * The post the files are attached to
+ */
+    postId?: string;
+
+    /**
+ * The post the files are attached to
+ */
+    post?: Post;
+
+    /**
+ * Set whether to show this modal or not
+ */
+    show?: boolean;
```

Comment:
Yeah but I have added `React.ComponentClass` as casting in `index.ts` and it was showing the error and If I remove the casting then it is showing some component type doesn't match any error.

Reply:
you shouldn't have to typecast it to React.ComponentClass, from a quick check I believe there is some typing error in the mapStateToProps probably because the props getting into the function and the ones being mapped doesn't match.

as a simple example, take a look at the `post_emoji` component